pkg/problemdetector/problem_detector.go

```diff
@@ -53,21 +53,32 @@ func (p *problemDetector) Run() error {
 		if err != nil {
 			// Do not return error and keep on trying the following config files.
 			glog.Errorf("Failed to start problem daemon %v: %v", m, err)
-			failureCount += 1
+			failureCount++
 			continue
 		}
 		if ch != nil {
 			chans = append(chans, ch)
 		}
 	}
-	if len(p.monitors) == failureCount {
+	allMonitors := p.monitors
+
+	if len(allMonitors) == failureCount {
 		return fmt.Errorf("no problem daemon is successfully setup")
 	}
+
+	defer func() {
```

Comment:
Just wondering, why is added. Was it not needed before?

Reply:
It didn't appear that NPD ever stopped the individual problem detectors. I added it because while trying to implement pause/continue it made sense to add. Although pause/continue was abandoned because it introduced more complexity and was not necessary. Functionally this logic shouldn't disrupt the critical path for NPD because it'll only run on shutdown of the service.

Reply:
Thanks for the info